packages/shared/src/common/date.ts

```diff
@@ -108,3 +109,66 @@ export function sortDate(
     ? buildDayjs(date1).diff(buildDayjs(date2))
     : buildDayjs(date2).diff(buildDayjs(date1));
 }
+
+export function throwIfInvalidHl7Timestamp(input: string): boolean {
```

Comment:
nit: this could be used in other scenarios, I'd call this something like `throwIfInvalidIsoDateWithoutFormatting()` or similar.

BTW, the format `YYYYMMDDHHMMSS` is still [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601)

Reply:
It's also ISO 8601, see the link above.